internal/repository/employee/employee_mysql.go

```diff
@@ -0,0 +1,132 @@
+package employee
+
+import (
+	pkgErrors "ProyectoFinal/pkg/errors"
+	"ProyectoFinal/pkg/models"
+	"database/sql"
+	"errors"
+)
+
+type mysqlRepository struct {
+	db *sql.DB
+}
+
+func NewMySQLRepository(db *sql.DB) Repository {
+	return &mysqlRepository{
+		db: db,
+	}
+}
+
+// GetAll executes a query to retrieve all employee records from the database,
+// scanning each row into Employee structs and returning them as a slice.
+// Returns an error if the query fails or if there's an issue scanning the results.
+func (r *mysqlRepository) GetAll() ([]models.Employee, error) {
+	rows, err := r.db.Query(QueryGetAllEmployees)
+	if err != nil {
+		return nil, pkgErrors.HandleMysqlError(err)
+	}
+	defer rows.Close()
+
+	if err = rows.Err(); err != nil {
+		return nil, pkgErrors.HandleMysqlError(err)
+	}
+
+	var employees []models.Employee
+	for rows.Next() {
+		var employee models.Employee
+
+		err := rows.Scan(&employee.ID, &employee.CardNumberID, &employee.FirstName, &employee.LastName, &employee.WarehouseID)
+		if err != nil {
+			return nil, pkgErrors.HandleMysqlError(err)
+		}
+
+		employees = append(employees, employee)
+	}
+
+	return employees, nil
+}
+
+// GetById performs a database lookup to find an employee by their unique identifier.
+// Returns the employee record if found, or a not found error if the ID doesn't exist.
+func (r *mysqlRepository) GetById(id int) (models.Employee, error) {
+	row := r.db.QueryRow(QueryGetEmployeeById, id)
+
+	var employee models.Employee
+
+	err := row.Scan(&employee.ID, &employee.CardNumberID, &employee.FirstName, &employee.LastName, &employee.WarehouseID)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return models.Employee{}, pkgErrors.WrapErrNotFound("employee", "id", id)
+		}
+		return models.Employee{}, pkgErrors.HandleMysqlError(err)
+	}
+
+	return employee, nil
+}
+
+// Create inserts a new employee record into the database and automatically assigns
+// the generated primary key ID back to the employee object.
+func (r *mysqlRepository) Create(employee *models.Employee) error {
+	result, err := r.db.Exec(QueryCreateEmployee, employee.CardNumberID, employee.FirstName, employee.LastName, employee.WarehouseID)
+	if err != nil {
+		return pkgErrors.HandleMysqlError(err)
+	}
+
+	lastInsertID, err := result.LastInsertId()
+	if err != nil {
+		return pkgErrors.HandleMysqlError(err)
```

Comment:
Desde el repo solo estamos devolviendo el err

La funcion HandleError ( la que usa el handler ) ya tiene la logica para identificar si es un err de mysql


```suggestion
		return pkgErrors.HandleMysqlError(err) -> return err
```